sync/client/leaf_syncer.go

```diff
@@ -48,12 +52,11 @@ type LeafClient interface {
 }
 
 // NewCallbackLeafSyncer creates a new syncer object to perform leaf sync of tries.
-func NewCallbackLeafSyncer(client LeafClient, tasks <-chan LeafSyncTask, requestSize uint16) *CallbackLeafSyncer {
+func NewCallbackLeafSyncer(client LeafClient, tasks <-chan LeafSyncTask, config *LeafSyncerConfig) *CallbackLeafSyncer {
```

Comment:
The name suggest `Callback` but it's a bit weird to take the callback in the config. Can we revert the config and take full params instead?